src/ripple/nodestore/impl/DatabaseShardImp.cpp

```diff
@@ -1350,19 +1500,106 @@ DatabaseShardImp::getCache(std::uint32_t seq)
     return std::make_pair(pCache, nCache);
 }
 
-std::uint64_t
-DatabaseShardImp::available() const
+bool
+DatabaseShardImp::sufficientStorage(
+    std::uint32_t numShards,
+    PathDesignation pathDesignation,
+    std::lock_guard<std::mutex> const&) const
 {
     try
     {
-        return boost::filesystem::space(dir_).available;
+        std::vector<std::uint64_t> capacities;
+
+        if (pathDesignation == PathDesignation::historical &&
+            !historicalPaths_.empty())
+        {
+            capacities.reserve(historicalPaths_.size());
+
+#if !BOOST_OS_LINUX
+            // Map byte counts to each path that
+            // shares that byte count.
+            std::unordered_map<std::uint64_t, std::vector<std::string>>
+                uniqueCapacities(historicalPaths_.size());
+#endif
+
+            // Get the available storage for each historical path
+            std::for_each(
```

Comment:
What is your criteria for using `std::for_each` vs. a range based for loop? Most of rippled uses range based for loops (and I prefer them as well). Unless there's a non-stylistic reason for this, I'd prefer we stick to range based for loops. (To be clear, there's nothing wrong with `std::for_each`, it just seems like there's a better way to do that now that we have range based for loops).

Reply:
Done